client/my-sites/plugins/controller.js

```diff
@@ -74,7 +75,8 @@ function renderSinglePlugin( context, siteUrl, isWpcomPlugin ) {
 			isWpcomPlugin: isWpcomPlugin,
 			onPluginRefresh: title => titleActions.setTitle( title )
 		} ),
-		document.getElementById( 'primary' )
+		document.getElementById( 'primary' ),
+		context.store
```

Comment:
> This is NOT a react context

thanks @artpi - I don't really care if it's React's `context` or some global `context` - more concerned about the case (if you will allow me to exaggerate for the sake of argument) where we build some widget that only depends on a few props but instead of making it just that we end up connecting that component to the store.

while the component proper is still pure and dandy, it gets called by the store on every update and we get those crazy re-render chains again plus the overhead of tracking all the observable references that the store is handling for us

please know that most of my :question: here is coming from ignorance and not from strong opinion